### PR TITLE
Install foundry toolchain in github workflow for integration tests

### PR DESCRIPTION
**Pull request by snormore**
Fixes failing GH workflow https://github.com/xmtp/didethresolver/actions/runs/7449494277/job/20266249707

### CODE REVIEW

File: .github/workflows/github-pages.yml [hunk 1]
```diff
@@ -37,6 +37,9 @@ jobs:
           profile: minimal
           override: true
 
+      - name: Install Foundry
```
Comment (snormore) [inline .github/workflows/github-pages.yml:40]: Alternatively I guess we could build the dev docker image and run tests through that

**Comment by snormore** [comment]
Should we just be running these tests via the dev docker image instead? 🤔

File: .github/workflows/github-pages.yml [hunk 2]
```diff
@@ -37,6 +37,9 @@ jobs:
...
+        uses: foundry-rs/foundry-toolchain@v1
```
Comment (snormore) [inline .github/workflows/github-pages.yml:41]: Should we just be running these tests via the dev docker image instead? 🤔

Comment (jac18281828) [reply .github/workflows/github-pages.yml:41]: this is the pages site generation so it requires a bit more work to generate the pages in docker...can be done but this is probably easier for now.

**Review by jac18281828 — APPROVED** [review]

**Pull request merged**